README.md

```diff
@@ -18,4 +18,129 @@ Also you have to configure the `github` maven repository. You can either include
+## Example
+First, when we implement events sourcing pattern we have to define aggregates. In this example we will be having user aggregate.
+```kotlin
+@AggregateType(aggregateEventsTableName = "aggregate-user")
+data class UserAggregate(
+    override val aggregateId: String
+) : Aggregate {
+    override var createdAt: Long = System.currentTimeMillis()
+    override var updatedAt: Long = System.currentTimeMillis()
+
+    var userName: String = ""
+    var userLogin: String = ""
+    var userPassword: String = ""
+    lateinit var defaultPaymentId: UUID
+    lateinit var defaultAddressId: UUID
+    var paymentMethods = mutableMapOf<UUID, PaymentMethod>()
+    var deliveryAddresses = mutableMapOf<UUID, DeliveryAddress>()
+}
+```
+@AggregateType annotations makes library understand that this is the aggregate recognisable by library.
+Each aggregate has its own domain events. They're written in UserAggregateDomainEvents.kt.
+This is the example of one of the events:
+```kotlin
+@DomainEvent(name = USER_CREATED_EVENT)
+class UserCreatedEvent(
+    val userLogin: String,
+    val userPassword: String,
+    val userName: String,
+    val userId: String,
+    createdAt: Long = System.currentTimeMillis(),
+) : Event<UserAggregate>(
+    name = USER_CREATED_EVENT,
+    createdAt = createdAt,
+    aggregateId = userId,
+) {
+    override fun applyTo(aggregate: UserAggregate) {
+        aggregate.userLogin = userLogin
+        aggregate.userPassword = userPassword
+        aggregate.userName = userName
+    }
+}
+```
+@DomainEvent annotation is also from the library. It accepts name of the event.
+Each event accepts some parameters from the constructor, returns event that belongs to the aggregate:
+```kotlin
+Event<UserAggregate>
+```
+and has override method applyTo that does manipulations with aggregate state.
+This is how this event looks in database:
+![](Example1.png)
+Second important concept of this example are subscribers. We can do subscriptions 2 ways.
+First one is this:
+```kotlin
+@Service
+@AggregateSubscriber(aggregateClass = UserAggregate::class, subscriberName = "demo-user-stream")
+class AnnotationBasedUserEventsSubscriber {
+    val logger: Logger = LoggerFactory.getLogger(AnnotationBasedUserEventsSubscriber::class.java)
+
+    @SubscribeEvent
+    fun userCreatedSubscriber(event: UserCreatedEvent) {
+        logger.info("User created {}", event.userName)
+    }
+}
+```
+As can be seen from the name of the class, it's annotation based. 
+It uses @AggregateSubscriber annotation, and listens to specific aggregate type catching it events.
+In the example we catch UserCreatedEvent and write it down in the logger.
+The second way of doing subscriptions will be shown in the next chapter that is about Projections.
+
+Another important aspect of example is projections usage. 
+Aggregates are mostly used for writing and changing information. In order to present information we use concept of projections.
```

Comment:
projection/view - synonims, lets designate it

Reply:
Did that in the readme